### ISSUE TITLE
[BUG]: Lack of recovery from meross cloud loss

### ISSUE THREAD

**Issue by rossisbudda** [theyosh/TerrariumPI#946]
**Setup:**
 - TerrariumPI version: x4.10.2
 - Install: Manual
 - Raspberry PI: 3B+
 - OS: Raspbian

Last night at around 04:50 one of my heating relays remained off when it should have turned back on. We lost connection to meross cloud then some other errors occurred. Resulting in a cold period. I can see in the logs that once we regained connection to the cloud the relay did not ever come back to life as such.

I think I will switch to a wired device as I can't rely on any cloud service, but thought you wold like to know about this as it seems that we could recover better.

I might have preferred a more robust recovery and regular re-checking of relay states to prevent long periods of stuck inactivity.

![image](https://github.com/user-attachments/assets/6c11663b-ebb7-4648-b1dd-e72ba4d1c70e)

[terrariumpi.log](https://github.com/user-attachments/files/16916728/terrariumpi.log)

 Thanks for looking. As I say I will switch to a wired relay and forget wifi products as I suppose they have an unreliability in their connectivity, or is there a more reliable wifi relay?

**Comment by theyosh** [comment]
Hmm, this is meross cloud related. Not wifi. There was some disconnect, which could be Wifi. But the meross code is rather complex, and does not work easy with my software.

Also, how long are you using this meross hardware, and how often did it happen before. I do have the idea that the Meross system is pretty stable. I do not get much issues with it.

Because what I see, is that the temperature stays at 24C in the logs, while the graph goes down. Which I cannot explain.
So somehow, the state of sensors and relays looks frozen in the system/code. And that I cannot explain. This is so hard to debug due to the fact Meross does not officially allow us to do this. But as long as we play nice with their api, they allow us to use their api with our custom code.

My question is, how did the temperature got restored. You just restarted TerrariumPI?

I think I cannot do much here. This is so hard to debug. And when it only happens once a year, it can live with it.

**Comment by rossisbudda** [comment]
Thanks so much. I will look out for it in the future and report if it is
too frequent.

I haven't been using this hardware for very long yet and I'm not certain
this has occurred before.

To "unstick" the relay I clicked the toggle button on the relays page, this
seemed to revive it. As this relay is frequently adjusting it is easy to
get stuck in a high or low state and I want to really avoid fluctuations in
the terrarium.  I've decided to try adding sudo service terrariumpi restart
to a crontab that runs every so often in the hope this would "refresh" the
engine.

Appreciate the advice 👍😎🙏
